oxo.py

```diff
@@ -1,32 +1,188 @@
+            self.player = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
-        """ If there are still empty squares on the board, return False.
-            If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        """
+        If there are still empty squares on the board, return False.
+        If there are no empty squares, return True.
+
+        Returns:
+            bool: False is board is not yet full. True if board is full.
+        """
+
+        for x in self.board[0]:
+            for y in self.board[1]:
+                if self.board[x][y] == 0:
+                    return False
+
+        return True
+
+    def collect_positions(self, player):
+        """Collects all the positions owned by a player
+
+        Args:
+            player (int): The player to check positions of.
+
+        Returns:
+            played_spaces (list of int): A list of the spaces owned by the specified player in the format: 0, 1...8 (Right to left, top to bottom).
+        """
+
+        played_spaces = []
+
+        for x in range(0, 3):
+            for y in range(0, 3):
+                if self.board[x][y] == player:
+                    position = [x, y]
+
+                    played_spaces.append(self.convert_to_1d_index(position, 3))
+
+        return played_spaces
+
+    def convert_to_1d_index(self, index, height):
+        """
+        Convert the indexes of a 2d array into a single index for a 1d array. Used to convert the x, y of board space into a single position.
+
+        Args:
+            index (list of int): The x, y indexes stored in a list.
+            height (int): The y dimension of the 2d array.
+
+        Returns:
+            new_index (int): The new index for a 1d array.
+        """
+
+        x = index[0]
+        y = index[1]
+
+        new_index = x + (y * height)
+        return new_index
+
+    def compare_lists(self, master_list, list_to_compare):
+        """
+        Check to see if a master list contains all the same items as the list being compared.
+
+        Args:
+            master_list (list): The main list.
+            list_to_compare (list): The list to compare to master.
+
+        Returns:
+            (bool): The result of a set comparison. True if master_list contains all elements in list_to_compare.
+        """
+
+        return set(list_to_compare) <= set(master_list)
+
+    def evaluate_board(self, player):
+        """
+        Evaluates the game board to see how many wins a given player has.
+
+        Args:
+            player (int); The player to evaluate for.
+
+        Returns:
+            number_of_wins (int): The number of wins the player has.
+        """
+
+        number_of_wins = 0
+        played_spaces = self.collect_positions(player)
+
+        for i in range (0, 8):
+            if self.compare_lists(played_spaces, self.winning_boards[i]) == True:
```

Comment:
Nice! Not sure if this is the most efficient way of checking for a winning line but it is quite neat